nemo_aligner/algorithms/dpo.py

```diff
@@ -308,27 +308,19 @@ def load_state_dict(self, state_dict):
     def augment_dataloader(self, dataloader):
         """Augment dataloader with ref policy log prob"""
         iter_dataloader = iter(dataloader)
-        buffer = []
         done = False
         while not done:
             try:
                 batch = next(iter_dataloader)
+                logprobs = self.model.get_ref_policy_logprobs(batch).cpu()
+                chosen_logps, reject_logps = torch.split(logprobs, len(logprobs) // 2, dim=0)
+                batch["ref_policy_log_probs_chosen"] = chosen_logps
+                batch["ref_policy_log_probs_rejected"] = reject_logps
+
+                yield batch
+                del logprobs, chosen_logps, reject_logps
             except StopIteration:
                 done = True
```

Comment:
While you're at it, can you get rid of the `done` flag and just `break` here?

Reply:
Done, please have a look